File: 003_Longest_Substring_Without_Repeating_Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> hashtable(255, -1);
        int result = 0;
        int i = -1;
        for (int j = 0; j < s.size(); j++) {
            i = std::max(hashtable[s[j]], i);
            hashtable[s[j]] = j;
            result = std::max(result, j - i);
        }
        return result;
    }
};

File: 003_Longest_Substring_Without_Repeating_Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        hashtable = [-1] * 255
        i = -1
        res = 0
        for j, char in enumerate(s):
            i = max(hashtable[ord(s[j])], i)
            hashtable[ord(s[j])] = j
            res = max(res, j - i)
        return res